packages/zoe/docs/zoe-zcf.md

```diff
@@ -0,0 +1,180 @@
+user can call `E(zoe).install(bundle)` to deploy code on-chain and
+thus create a new "installation" for the contract code. An
+installation is an object that can be used for its identity to compare
+whether the code used in different contract instances is the same. In
+the future
+([Issue](https://github.com/Agoric/agoric-sdk/issues/3871)), a user
+will be able to get the hash from an installation, so that a user can
+compare the hash with the hash of code hosted off-chain.
+
+Anyone with an `installation` can ask the Zoe Service to create a new
+instance of a contract with a particular assignment of keywords to
+Brands, and specified terms. When the Zoe Service receives the
+`E(zoe).startInstance()` call, it creates a new vat (getting the vat's
+root object in return) and a new object `zoeInstanceAdminForZcf` which
+serves as the ZCF vat's main connection to Zoe. 
+
+Zoe then calls `E(zcfRoot).executeContract()` to start the contract.
+The parameters include a promise for the Zoe Service, the source
+bundle, the invitation Issuer, `zoeInstanceAdminForZcf`, the instance
+record, issuer records, any private arguments, and the fee purse.
+
+The code in the ZCF Vat evaluates the contract bundle, makes a ZCF
+object for the contract's use, and calls the contract's `start`
+method. The contract can return any or all of the following: a
+`publicFacet` (accessible to anyone with the contract instance), a
+`creatorFacet` (an ocap only given to the owner of a contract), and a
+`creatorInvitation` (an invitation only given to the owner of a
+contract). ZCF passes these on from the contract to Zoe and Zoe
+returns them to the caller of `E(zoe).startInstance()`. ZCF also
+returns to Zoe an object solely for communicating an offer to ZCF:
+`handleOfferObj`.
+### Making an Invitation and Exercising an Offer
+
+The Zoe Service creates and holds the mint for invitations. The Zoe
+Service freely shares the corresponding issuer, so everyone can verify
+invitations.
+
+When an invitation is exercised (`E(zoe).offer(invitation, ...)`), Zoe
+burns the invitation, escrows the payments, calls
+`E(handleOfferObj).handleOffer`, which informs ZCF that a new offer
+has been made, and returns a newly created `userSeat`. `handleOffer`
+in ZCF calls the `offerHandler`, the particular function that the
+contract code registered when it called `zcf.makeInvitation`. ZCF
+records what it needs to know about the offer, then returns two things
+to Zoe: a promise for the offerResult (the return value from the
+`offerHandler`) and an `exitObj` that Zoe saves for later use.
+
+In order to ensure that the `zcf` object in the contract vat has the
+most up-to-date information about the seats that have exited
+synchronously accessible, if a user calls `userSeat.tryExit()`, the
+Zoe Service calls `E(exitObj).exit()`, making sure that even if the
+user originates the request, the exit process always goes through the
+contract vat first.
+
```

Comment:
```suggestion
```suggestion
In order to ensure that the `zcf` object in the contract vat has
synchronous access to the most up-to-date information about
seats that have exited, if a user calls `userSeat.tryExit()`, the
Zoe Service calls `E(exitObj).exit()`. This ensures that even when the
user originates the request, the exit process always goes through the
contract vat first.

```